File: frontend/src/features/usergroup/http/api/CreateUserGroup.ts
import type { CreateUserGroupRequest } from '@/features/usergroup/http/request/CreateUserGroupRequest'
import type { UserGroupDetail } from '@/features/usergroup/http/response/UserGroupDetail'
import {
  fromUserGroupDetailContract,
  toCreateUserGroupRequestContract,
} from '@/features/usergroup/http/codec/UserGroupHttpCodecs'
import { postJson } from '@/shared/api/http-client'

export async function createUserGroup(request: CreateUserGroupRequest): Promise<UserGroupDetail> {
  return postJson('/api/user-groups', fromUserGroupDetailContract, toCreateUserGroupRequestContract(request))
}

File: backend/src/main/scala/domains/usergroup/http/api/CreateUserGroup.scala
package domains.usergroup.http.api



import domains.usergroup.http.*
import domains.usergroup.http.codec.UserGroupHttpCodecs.given
import cats.effect.IO
import domains.usergroup.application.UserGroupCommands
import domains.usergroup.application.input.{CreateUserGroupRequest}
import org.http4s.HttpRoutes
import org.http4s.circe.CirceEntityCodec.*
import org.http4s.dsl.Http4sDsl
import org.http4s.dsl.io.*

object CreateUserGroup:

  def routes(context: UserGroupHttpRouteContext)(using Http4sDsl[IO]): HttpRoutes[IO] =
    HttpRoutes.of[IO] {
      case request @ POST -> Root / "api" / "user-groups" =>
        context.handlers.executeDecoded[CreateUserGroupRequest, CreateUserGroupRequest, UserGroupCommands.CreateUserGroupResult](
          request,
          UserGroupHttpPlanDefinitions.createUserGroup
        )(identity)
    }